cxx-squid/src/main/java/org/sonar/cxx/preprocessor/ExpressionEvaluator.java

```diff
@@ -440,11 +443,16 @@ BigInteger decode(String number) {
     int radix = 10;
     if (number.length() > 2) {
       if (number.charAt(0) == '0') {
-        if (number.charAt(1) == 'x' || number.charAt(1) == 'X') {
-          radix = 16; // 0x...
-          number = number.substring(2);
-        } else {
           radix = 8; // 0...
+        try {
```

Comment:
Would not use an exception to distinguish between octal and hex. 'Good case' handling over exceptions is always expensive. Why not keep looking for the 'x'?